From 802b2c4eb0be853f3116f4014daf7222299e7323 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:00 -0400
Subject: [net] usb/cdc_ncm: remove descriptor pointers

Message-id: <1410887463-20674-169-git-send-email-dzickus@redhat.com>
Patchwork-id: 95156
O-Subject: [RHEL7 PATCH 168/591] net: cdc_ncm: remove descriptor pointers
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 832922362e1308aaef95a43383577d56f51fbc3c
Author: Bjorn Mork <bjorn@mork.no>
Date:   Fri Nov 1 11:16:47 2013 +0100

    net: cdc_ncm: remove descriptor pointers

    header_desc was completely unused and union_desc was never used
    outside cdc_ncm_bind_common.

    Cc: Alexey Orishko <alexey.orishko@gmail.com>
    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/usb/cdc_ncm.c b/drivers/net/usb/cdc_ncm.c
index 4de3a54..89280ec 100644
--- a/drivers/net/usb/cdc_ncm.c
+++ b/drivers/net/usb/cdc_ncm.c
@@ -371,6 +371,7 @@ static const struct ethtool_ops cdc_ncm_ethtool_ops = {
 
 int cdc_ncm_bind_common(struct usbnet *dev, struct usb_interface *intf, u8 data_altsetting)
 {
+ const struct usb_cdc_union_desc *union_desc = NULL;
  struct cdc_ncm_ctx *ctx;
  struct usb_driver *driver;
  u8 *buf;
@@ -405,16 +406,15 @@ int cdc_ncm_bind_common(struct usbnet *dev, struct usb_interface *intf, u8 data_
 
   switch (buf[2]) {
   case USB_CDC_UNION_TYPE:
-   if (buf[0] < sizeof(*(ctx->union_desc)))
+   if (buf[0] < sizeof(*union_desc))
     break;
 
-   ctx->union_desc =
-     (const struct usb_cdc_union_desc *)buf;
+   union_desc = (const struct usb_cdc_union_desc *)buf;
 
    ctx->control = usb_ifnum_to_if(dev->udev,
-     ctx->union_desc->bMasterInterface0);
+             union_desc->bMasterInterface0);
    ctx->data = usb_ifnum_to_if(dev->udev,
-     ctx->union_desc->bSlaveInterface0);
+          union_desc->bSlaveInterface0);
    break;
 
   case USB_CDC_ETHERNET_TYPE:
@@ -457,7 +457,7 @@ advance:
  }
 
  /* some buggy devices have an IAD but no CDC Union */
- if (!ctx->union_desc && intf->intf_assoc && intf->intf_assoc->bInterfaceCount == 2) {
+ if (!union_desc && intf->intf_assoc && intf->intf_assoc->bInterfaceCount == 2) {
   ctx->control = intf;
   ctx->data = usb_ifnum_to_if(dev->udev, intf->cur_altsetting->desc.bInterfaceNumber + 1);
   dev_dbg(&intf->dev, "CDC Union missing - got slave from IAD\n");
diff --git a/include/linux/usb/cdc_ncm.h b/include/linux/usb/cdc_ncm.h
index 89b52a0..cad54ad 100644
--- a/include/linux/usb/cdc_ncm.h
+++ b/include/linux/usb/cdc_ncm.h
@@ -92,9 +92,7 @@ struct cdc_ncm_ctx {
  struct tasklet_struct bh;
 
  const struct usb_cdc_ncm_desc *func_desc;
- const struct usb_cdc_mbim_desc   *mbim_desc;
- const struct usb_cdc_header_desc *header_desc;
- const struct usb_cdc_union_desc *union_desc;
+ const struct usb_cdc_mbim_desc *mbim_desc;
  const struct usb_cdc_ether_desc *ether_desc;
 
  struct usb_interface *control;
-- 
1.7.1